### ISSUE TITLE
Multiple ingress class values

### ISSUE THREAD

**Issue by erwbgy** [projectcontour/contour#4097]
**Allow the contour --ingress-class-name value to be a comma-separated list of classes to match against**

We run multiple Contour instances in each of our clusters for different groups with each instance having a different ingress class value.  In less busy clusters this results in wasted resources with many instances largely idle so we'd like to be able to share a single Contour instance but have it watch objects for multiple ingress classes.  This way users can continue to use their own ingress classes and not have to reconfigure when deploying to different clusters but we avoid running multiple instances when we don't need to.

For example:
```
  --ingress-class-name=contour,team1,team2
```
would process HTTPProxy or Ingress objects that have an ingress.class annotation or ingressClassName field with any of these values.

If you're open to this change it is a relatively small change to the matches function in internal/ingressclass/ingressclass.go (excluding tests) and I can create a pull request.

**Comment by youngnick** [comment]
It seems like this is because users want to reuse the same Ingress object in multiple clusters, that may have different Contour setups installed. So you want to have the users be able to use the same, unmodified Ingress object wherever, and have the ingress-class magically hook up to the right Contour depending on where you're running.

Having a way to have Contour listen to multiple ingress-classes makes sense, I guess. I wonder how this would work if you are using the IngressClass object as well though...

tl;dr I can't see any issues off the top of my head, but it would be nice to have a consistent story for this across different ways of configuring ingress-class. I'll also keep this in mind for Gateway and GatewayClass for the future.
